job-runner/runner/runner.go

```diff
@@ -133,9 +141,19 @@ TRY_LOOP:
 			Stdout:      jobRet.Stdout,
 			Stderr:      jobRet.Stderr,
 		}
-		if err := r.rmc.CreateJL(r.reqId, jl); err != nil {
-			tryLogger.Errorf("problem sending job log (%#v) to the RM: %s", jl, err)
+		// Send the JL to the RM.
+		err := retry.Do(JOB_LOG_TRIES, JOB_LOG_RETRY_WAIT,
+			func() error {
+				return r.rmc.CreateJL(r.reqId, jl)
+			},
+			func(err error) {
+				tryLogger.Errorf("problem sending job log (%#v) to the RM: %s. Retrying...", jl, err)
```

Comment:
I wouldn't log this either. We don't care about transient failures. Logging the final error is good enough.